update/transaction.go

```diff
@@ -0,0 +1,70 @@
+package update
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/fabric8-services/fabric8-common/log"
+	"github.com/pkg/errors"
+)
+
+const TenantsUpdateAdvisoryLockID = 4242
+
+func Transaction(db *sql.DB, lockAndDo func(tx *sql.Tx) error) error {
+	var err error
+
+	if db == nil {
+		return fmt.Errorf("Database handle is nil\n")
+	}
+
+	var tx *sql.Tx
+
+	tx, err = db.Begin()
+	if err != nil {
+		return errors.Wrap(err, "failed to start transaction")
+	}
+
+	err = lockAndDo(tx)
```

Comment:
Let's also add a lock timeout in the beginning of transaction. I guess something like `SET LOCAL lock_timeout = 'XXXs';` where XXX is number of seconds we want to set as the lock timeout. Just to be on the safe side :)

Reply:
It's part of b0721b0 - I set the timeout to one minute. If you want to ask me why this duration - there is no big reason, maybe just that I didn't want to use something too short nor something too long :-)